stdlib/tests/crypto/stark/verifier_recursive/mod.rs

```diff
@@ -24,6 +24,62 @@ pub struct VerifierData {
     pub advice_map: Vec<(Word, Vec<Felt>)>,
 }
 
+/// Generates advice inputs required for recursive STARK proof verification in Miden VM.
+///
+/// During the course of its execution, the recursive verifier will make use of non-deterministic
+/// advice data through the advice provider.
+/// In what follows is a description of the expected layout per component of the advice provider.
+/// We also include the expected layout of the operand stack for completness.
+///
+/// # Operand Stack Layout
+///
+/// ```
+/// | Position | Content           | Size (in Felt) | Description                      |
+/// | -------- | ----------------- | -------------- | -------------------------------- |
+/// | 0        | grinding_factor   | 1              | Proof-of-work difficulty in bits |
+/// | 1        | num_queries       | 1              | Number of FRI queries            |
+/// | 2        | trace_length_log2 | 1              | Log2 of execution trace length   |
+/// ```
+///
+/// # Advice Stack Layout
+///
+/// ```
+/// | Position | Content                | Size (in Felt) | Description                                           |
+/// | -------- | ---------------------- | -------------- | ----------------------------------------------------- |
+/// | 0        | variable_len_pi_size   | 1              | Size of variable length PI in Felt                    |
+/// | 1-n      | public_inputs_data     | varies         | Input/output stacks + Program digest + kernel digests |
+/// | n+1-n+4  | aux_randomness         | 4              | β = (β₀, β₁), ɑ = (ɑ₀, ɑ₁) (filled later)             |
```

Comment:
What does "filled later" mean here? (and same question for line 55 below).

Reply:
What I was trying to say, in a succinct way, is that these values are out-of-order with respect to the interaction transcript. Thus, in the advice tape, we push place holders for these values and only later on in the interaction can we fill these place holders with the appropriate values.
Probably "filled late" is just confusing without enough context, so I have just removed it

Reply:
What does "placeholder" mean in this context? My understanding was that we put real values onto the advice stack, but then use the "optimistically" - and only later confirm that the values we used were indeed derived correctly. Or does it mean something else?

Reply:
Just on the Rust size. We basically, push say a `[0 ,0]` and record the index before this push. Later on when we can generate the random challenge, we can do something like `advice_tape[index] = alpha_0` and `advice_tape[index + 1] = alpha_1`.
Overall, this is just a Rust implementation detail i.e., it could be done differently